lib/spack/spack/installer.py

```diff
@@ -693,6 +699,17 @@ def __str__(self):
         return '{0}: {1}; {2}; {3}; {4}'.format(
             self.pid, requests, tasks, installed, failed)
 
+    def _set_term_title(self, text, force=False):
+        if not spack.config.get('config:set_terminal_title', False):
+            return
+
+        if not sys.stdout.isatty() and not force:
+            return
+
+        status = '{0} [{1}/{2}]'.format(text, self.pkg_num, self.pkg_count)
+        sys.stdout.write('\033]0;Spack: {0}\007'.format(status))
+        sys.stdout.flush()
```

Comment:
I think this functionality would be better as a small helper class rather than as a method with class data.  Maybe something like:
```python
class TermWriter(object):
    def __init__(self, pkg_count):
        self.pkg_num = 0
        self.pkg_count = pkg_count

    def next_package(self):
        self.pkg_num += 1

    def set_title(self, text):
        status = '{0} [{1}/{2}]'.format(text, self.pkg_num, self.pkg_count)
        sys.stdout.write('\033]0;Spack: {0}\007'.format(status))
        sys.stdout.flush()


class NullopTermWriter(object):
    def next_package(self):
        pass

    def set_title(self):
        pass


def make_term_writer(pkg_count):
    if (spack.config.get('config:set_terminal_title', False)
        and sys.stdout.isatty()):
        return TermWriter(pkg_count)

    # Terminal output disabled
    return NullopTermWriter()
```
the instantiate locally within `install`. It'll improve the ability to test and reduces class clutter.

Reply:
I've extracted it into a helper class now but kept it as a single class (I think we can afford the `if` within `set`).